add autocomplete API, fix link in common error (#62)
amadeus4dev/developer-guides: docs/resources/hotels.md

### Autocomplete Hotel Names 

> Your application can also display the list of autocompleted hotel names when users attempt to search hotels with keywords. 

Reviewer: ```suggestion
Your application can also display a list of suggested hotel names based on keywords used in the search query. 
```

> [Hotel Name Autocomplete API](https://developers.amadeus.com/self-service/category/hotel/api-doc/hotel-name-autocomplete/api-reference) will provide a list of up to 20 hotels whose names most closely match for a given text string. For each hotel result, the API also provides descriptive data including hotel name, address, geocode, property type, IATA hotel code and an Amadeus hotel ID. 

Reviewer: ```suggestion
[Hotel Name Autocomplete API](https://developers.amadeus.com/self-service/category/hotel/api-doc/hotel-name-autocomplete/api-reference) provides a list of up to 20 hotels whose names most closely match the search query string. For each hotel in the results, the API also provides descriptive data, including the hotel name, address, geocode, property type, IATA hotel code and the Amadeus hotel ID. 
```
